File: 1.1.py
#  Si un número X es par o impar

def solicitarNumero():
    numero = 0
    while numero == 0:
        try:
            numero = int(input("Ingresa un numero: "))
            return numero
        except ValueError:
            print("El valor ingresado no es un numero")


def comprobarParidad(numero):
    if numero % 2 == 0:
        print("El numero es par")
    else:
        print("El numero es impar")

def main():
    numero = solicitarNumero()
    comprobarParidad(numero)

if __name__ == "__main__":
    main()

File: 1.1.kt
/* Si un número X es par o impar */

fun solicitarNumero(): Int {
    var numero: Int? = null

    while (numero == null) {
        print("Ingresa un número: ")
        numero = readLine()?.toIntOrNull()

        if (numero == null) {
            println("Debe ser un número entero válido.")
        }
    }

    return numero
}

fun ComprobarParidad(numero : Int) {
    if (numero % 2 == 0) {
        println("El numero es Par")
    }
    else {
        println("El numero es Impar")
    }
}

fun main() {
    val numero = solicitarNumero()
    ComprobarParidad(numero)
}
